File: src/main/java/org/railwaystations/rsapi/adapter/in/task/NotifyUsersTask.java
package org.railwaystations.rsapi.adapter.in.task;

import lombok.RequiredArgsConstructor;
import org.railwaystations.rsapi.core.services.NotifyUsersService;
import org.springframework.scheduling.annotation.Scheduled;
import org.springframework.stereotype.Component;

@Component
@RequiredArgsConstructor
public class NotifyUsersTask {

    private final NotifyUsersService notifyUsersService;

    @Scheduled(cron = "0 0 1 * * *")
    public void notifyUsers() {
        notifyUsersService.notifyUsers();
    }

}

File: adapter/src/main/kotlin/org/railwaystations/rsapi/adapter/task/NotifyUsersTask.kt
package org.railwaystations.rsapi.adapter.task

import org.railwaystations.rsapi.core.ports.inbound.NotifyUsersUseCase
import org.springframework.scheduling.annotation.Scheduled
import org.springframework.stereotype.Component

@Component
class NotifyUsersTask(private val notifyUsersUseCase: NotifyUsersUseCase) {

    @Scheduled(cron = "0 0 1 * * *")
    fun notifyUsers() {
        notifyUsersUseCase.notifyUsers()
    }
}
